audit/entry_formatter.go

```diff
@@ -107,33 +119,36 @@ func (f *EntryFormatter) Process(ctx context.Context, e *eventlogger.Event) (*ev
 		data.Request.Headers = adjustedHeaders
 	}
 
+	// Using any as we have two different types that we can get back from either
+	// FormatRequest or FormatResponse.
+	var entry any
 	var result []byte
 
 	switch a.Subtype {
 	case RequestType:
-		entry, err := f.FormatRequest(ctx, data)
-		if err != nil {
-			return nil, fmt.Errorf("%s: unable to parse request from audit event: %w", op, err)
-		}
-
-		result, err = jsonutil.EncodeJSON(entry)
-		if err != nil {
-			return nil, fmt.Errorf("%s: unable to format request: %w", op, err)
-		}
+		entry, err = f.FormatRequest(ctx, data)
 	case ResponseType:
-		entry, err := f.FormatResponse(ctx, data)
-		if err != nil {
-			return nil, fmt.Errorf("%s: unable to parse response from audit event: %w", op, err)
-		}
-
-		result, err = jsonutil.EncodeJSON(entry)
-		if err != nil {
-			return nil, fmt.Errorf("%s: unable to format response: %w", op, err)
-		}
+		entry, err = f.FormatResponse(ctx, data)
 	default:
 		return nil, fmt.Errorf("%s: unknown audit event subtype: %q", op, a.Subtype)
 	}
 
+	if err != nil {
+		return nil, fmt.Errorf("%s: unable to parse %s from audit event: %w", op, a.Subtype.String(), err)
+	}
+
+	// Exclude data from the RequestEntry/ResponseEntry.
+	m, err := f.excludeFields(entry)
+	if err != nil {
+		return nil, fmt.Errorf("%s: unable to exclude audit data from %s: %w", op, a.Subtype.String(), err)
```

Comment:
If we fail to do exclusions, is it better to fail the request altogether, or to proceed with writing an unmodified audit records?  I'm leaning toward the latter.

Reply:
🤔  I'm adding further checks to the errors we encounter within the `excludeFields` method, so that if we get one here, it really is a show stopper.

Still on the fence about whether we still carry on or block, so leaving this comment unresolved for the time being.

Reply:
Still relevant but lives here now: https://github.com/hashicorp/vault/pull/25003/files#diff-99656956333b4ad7e64221705191895c2bb140fd7d47c9f088d9520f781b8b40R148-R150